### PR TITLE
Update pipeline.md

### PR DESCRIPTION
**Pull request by macchiati**
A couple of clarifications in the PR prep section

### CODE REVIEW

File: docs/pipeline.md [hunk 1]
```diff
@@ -58,10 +58,10 @@ Indic scripts only:
 
 ---
 PR preparation:
-- [ ] Proposal document — Cite L2 number
-- [ ] UTC decisions — Cite _in extenso_
+- [ ] If a Proposal document — Cite L2 number in the format XXXX
+- [ ] If for a UTC decision — Cite in the format UTC-\d\d\d-[MC]\d+
```
Comment (markusicu) [inline docs/pipeline.md:62]: I think we want both of these, not either-or --> so I wouldn't add the "If" -- I would leave the parts before the em dashes as is.

File: docs/pipeline.md [hunk 2]
```diff
@@ -58,10 +58,10 @@ Indic scripts only:
 
 ---
 PR preparation:
-- [ ] Proposal document — Cite L2 number
-- [ ] UTC decisions — Cite _in extenso_
+- [ ] If a Proposal document — Cite L2 number in the format XXXX
```
Comment (markusicu) [inline docs/pipeline.md:61]:
```suggestion
- [ ] If a Proposal document — Cite L2 number in the format L2/yy-nnn
```

File: docs/pipeline.md (hunk 1 above)
Comment (eggrobin) [reply docs/pipeline.md:62]: Seconded.

File: docs/pipeline.md [hunk 3]
```diff
@@ -58,10 +58,10 @@ Indic scripts only:
 
 ---
 PR preparation:
-- [ ] Proposal document — Cite L2 number
-- [ ] UTC decisions — Cite _in extenso_
+- [ ] If a Proposal document — Cite L2 number in the format L2/yy-nnn
+- [ ] If for a UTC decision — Cite in the format UTC-\d\d\d-[MC]\d+
```
Comment (eggrobin) [inline docs/pipeline.md:62]:
```suggestion
- [ ] Proposal document — Cite L2 number in the format L2/yy-nnn
- [ ] UTC decisions — Cite in the format UTC-\d\d\d-[MC]\d+
  - unless preparing a PR for SAH recommendations not yet presented to the UTC.
```

File: docs/pipeline.md [hunk 4]
```diff
@@ -58,10 +58,10 @@ Indic scripts only:
...
 - [ ] data-for-new — Set label
-- [ ] pipeline-* — Set label
+- [ ] pipeline-* — Set label to **pipeline-provisionally-assigned** or **pipeline-`<version>`**, depending on whether provisional or not.
```
Comment (eggrobin) [inline docs/pipeline.md:64]:
```suggestion
- [ ] pipeline-* — Set label to **pipeline-sah-recs** if the characters are not yet in the pipeline, and **pipeline-provisionally-assigned**, or **pipeline-`<version>`** depending on their status in [the Pipeline](https://unicode.org/alloc/Pipeline.html#future).
```

**Review by eggrobin — APPROVED** [review]

**Pull request merged**